src/app/pcasts/utils/topic_utils.py

```diff
@@ -0,0 +1,69 @@
+import json
+from app import constants
+
+# Returns a mapping from (sub)topic name to least significant bit and (sub)topic
+# id to least significant bit.
+def create_topics_mappings():
+  topic_bit_position, subtopic_bit_position = 0, 0
+  topic_name_offset, subtopic_name_offset = {}, {}
+  topic_id_offset, subtopic_id_offset = {}, {}
+  name_to_id = {}
+  topics, subtopics = [], []
+
+  json_data = open(constants.TOPIC_FILE)
+  data = json.load(json_data)
+  for topic in data['topics']:
+    name_to_id[topic['name']] = topic['id']
+    topics.append(topic['name'])
+    for subtopic in topic['subtopics']:
+      name_to_id[subtopic['name']] = subtopic['id']
+      subtopics.append(subtopic['name'])
+
+  topics.sort()
+  subtopics.sort()
+  for topic in topics:
+    topic_name_offset[topic] = topic_bit_position
+    topic_id_offset[name_to_id[topic]] = topic_bit_position
+    topic_bit_position += 1
+
+  for subtopic in subtopics:
+    subtopic_name_offset[subtopic] = subtopic_bit_position
+    subtopic_id_offset[name_to_id[subtopic]] = subtopic_bit_position
+    subtopic_bit_position += 1
+
+  return topic_id_offset, subtopic_id_offset, \
+         topic_name_offset, subtopic_name_offset
+
+# Returns an id that represents both the old topics and the new topic
+# identified by subtopic_offset
+def create_topic_id(old_topic_id, topic_name):
```

Comment:
I think the naming here too is a little unintuitive, this function ORs the old bitmask with the one corresponding to the requested topic. Maybe something like merge_topic_masks. Also none of these functions mention that they deal with bit operations, it's easy to confuse a topic's id like 1309 and the equivalent bit id like 000100. We might want to make that distinction clearer with these names

Reply:
If the topic name is not valid, this function will just return `old_topic_id` without throwing or logging any errors, is that the intended behavior?

Reply:
I set up a reminder in slack to discuss this in person as this is not a functionality problem and we need to get this out.